sdk/src/main/scala/com/horizen/AbstractWallet.scala

```diff
@@ -102,6 +105,22 @@ abstract class AbstractWallet[
 
   def applyConsensusEpochInfo(epochInfo: ConsensusEpochInfo): W
 
+  def generateNextSecret[T <: Secret](secretCreator: SecretCreator[T]): Try[W] = Try {
+    require(secretCreator != null, "AbstractWallet: Secret creator must be NOT NULL.")
+    var nonce = this.secrets().size
+    val salt: Array[Byte] = secretCreator.salt()
+    while(true) {
```

Comment:
For improving safety and robustness, consider adding a limit on the number of attempts. Such limit can be the size of the list of secrets currently stored in the storage.

Reply:
Done